zypp/target/TargetImpl.cc

```diff
@@ -1896,7 +1948,8 @@ namespace zypp
 
       bool abort = false;
       zypp::AutoDispose<std::unordered_map<int, ManagedFile>> locCache([]( std::unordered_map<int, ManagedFile> &data ){
-        for ( auto &[key, value] : data ) {
+        for ( auto &[_, value] : data ) {
+          [[maybe_unused]] auto & key { _ };
```

Comment:
Does that silence a warning?  It seems GCC8 can correctly handle the [[maybe_unused]] directly in the for() declaration

Reply:
I guess the optimizer will remove taking the reference there, but it still looks not so nice

```
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wunused-variable"
    for (auto & [unused, val]: my_map)
#pragma GCC diagnostic pop
```

also is possible but is not exactly pretty

Reply:
Using `(void)_;` to silence g++ < 8